File: Java/LeetCode/Count Equal and Divisible Pairs in an Array.java
// https://leetcode.com/problems/count-equal-and-divisible-pairs-in-an-array/

class Solution {

    public int countPairs(int[] nums, int k) {
        int pairCount = 0;

        for (int i = 0; i < nums.length-1; ++i) {
            for (int j = i+1; j < nums.length; ++j) {
                if (nums[i] == nums[j] && (i*j % k == 0)) {
                    pairCount++;
                }
            }
        }

        return pairCount;
    }

}

File: Kotlin/LeetCode/Count Equal and Divisible Pairs in an Array.kt
// https://leetcode.com/problems/count-equal-and-divisible-pairs-in-an-array

class Solution {

    fun countPairs(nums: IntArray, k: Int): Int {
        var pairCount = 0;

        for (i in 0 until nums.size-1) {
            for (j in i+1 until nums.size) {
                if (nums[i] == nums[j] && (i*j % k == 0)) {
                    pairCount++;
                }
            }
        }

        return pairCount;
    }

}
